src/core/styles/StyledInputField.tsx

```diff
@@ -0,0 +1,265 @@
+  const { onChange, onEnter, label, value, required, placeholder, helperText, disabled, errorMessage, error } = props;
+  return (
+    <>
+      <StyledInputLabel><FormattedMessage id={label} /></StyledInputLabel>
+      <TextFieldRoot
+        fullWidth
+        disabled={disabled}
+        variant="outlined"
+        required={required}
+        placeholder={placeholder + ""}
+        value={value}
+        error={error}
+        onChange={({ target }) => onChange(target.value)}
+        onKeyPress={onEnter ? (event) => {
+          const key = event.key;
+          if (key === 'Enter') {
+            onEnter();
+          }
+        } : undefined}
+      />
+      <BottomText helperText={helperText} errorMessage={errorMessage} error={error} />
+    </>
+  );
+}
+
+const StyledNumberField: React.FC<StyledInputFieldProps<number>> = (props) => {
+  const { onChange, onEnter, label, value, required, placeholder, helperText, disabled, error, errorMessage } = props;
+  return (
+    <>
+      <StyledInputLabel><FormattedMessage id={label} /></StyledInputLabel>
+      <TextFieldRoot
+        fullWidth
+        disabled={disabled}
+        variant="outlined"
+        required={required}
+        type={"number"}
+        placeholder={placeholder + ""}
+        value={value}
+        error={error}
+        onChange={({ target }) => onChange(target.value as any)}
+        onKeyPress={onEnter ? (event) => {
+          const key = event.key;
+          if (key === 'Enter') {
+            onEnter();
+          }
+        } : undefined}
+      />
+      <BottomText helperText={helperText} errorMessage={errorMessage} error={error} />
+    </>
+  );
+}
+
+const StyledDateField: React.FC<StyledInputFieldProps<string>> = (props) => {
+  const { onChange, onEnter, label, value, required, placeholder, helperText, disabled, error, errorMessage } = props;
+  const placeholderColor = value ? 'text.primary' : 'text.secondary';
+  return (
+    <>
+      <StyledInputLabel><FormattedMessage id={label} /></StyledInputLabel>
+      <TextFieldRoot
+        fullWidth
+        disabled={disabled}
+        variant="outlined"
+        required={required}
+        type={"date"}
+        placeholder={placeholder + ""}
+        value={value}
+        error={error}
+        sx={{
+          '& .MuiInputBase-input': {
+            color: placeholderColor,
+          }
+        }}
+        onChange={({ target }) => onChange(target.value as any)}
+        onKeyPress={onEnter ? (event) => {
+          const key = event.key;
+          if (key === 'Enter') {
+            onEnter();
+          }
+        } : undefined}
+      />
+      <BottomText helperText={helperText} errorMessage={errorMessage} error={error} />
+    </>
+  );
+}
+
+const StyledDateTimeField: React.FC<StyledInputFieldProps<string>> = (props) => {
+  const { onChange, onEnter, label, value, required, placeholder, helperText, disabled, error, errorMessage } = props;
```

Comment:
`required` here appears to be unused.  Should it be a part of the `<TextFieldRoot />`  props ?